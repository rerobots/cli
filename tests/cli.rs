// SCL <scott@rerobots.net>
// Copyright (C) 2021 rerobots, Inc.

use assert_cmd::Command;


#[test]
fn prints_version() {
    let mut cmd = Command::cargo_bin("rerobots").unwrap();
    let assert = cmd.arg("-V").assert();
    assert
        .stdout(format!("{}\n", env!("CARGO_PKG_VERSION")))
        .success();
}


#[test]
fn prints_help() {
    let mut cmd = Command::cargo_bin("rerobots").unwrap();
    let assert = cmd.arg("help").assert();
    let output = assert.get_output().clone();
    assert.success();
    insta::assert_display_snapshot!(String::from_utf8(output.stdout).unwrap());

    // Alternative style: -h
    let mut cmd = Command::cargo_bin("rerobots").unwrap();
    let assert = cmd.arg("-h").assert();
    let output = assert.get_output().clone();
    assert.success();
    insta::assert_display_snapshot!("prints_help", String::from_utf8(output.stdout).unwrap());

    // Alternative style: --help
    let mut cmd = Command::cargo_bin("rerobots").unwrap();
    let assert = cmd.arg("--help").assert();
    let output = assert.get_output().clone();
    assert.success();
    insta::assert_display_snapshot!("prints_help", String::from_utf8(output.stdout).unwrap());
}


#[test]
fn prints_help_search() {
    let mut cmd = Command::cargo_bin("rerobots").unwrap();
    let assert = cmd.arg("help").arg("search").assert();
    let output = assert.get_output().clone();
    assert.success();
    insta::assert_display_snapshot!("prints_help_search", String::from_utf8(output.stdout).unwrap());

    // Alternative style: -h
    let mut cmd = Command::cargo_bin("rerobots").unwrap();
    let assert = cmd.arg("search").arg("-h").assert();
    let output = assert.get_output().clone();
    assert.success();
    insta::assert_display_snapshot!("prints_help_search", String::from_utf8(output.stdout).unwrap());

    // Alternative style: --help
    let mut cmd = Command::cargo_bin("rerobots").unwrap();
    let assert = cmd.arg("search").arg("--help").assert();
    let output = assert.get_output().clone();
    assert.success();
    insta::assert_display_snapshot!("prints_help_search", String::from_utf8(output.stdout).unwrap());
}
