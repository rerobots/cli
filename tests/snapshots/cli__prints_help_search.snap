---
source: tests/cli.rs
assertion_line: 47
expression: "String::from_utf8(output.stdout).unwrap()"
---
rerobots-search 
Search for matching deployments. empty query implies show all existing workspace
deployments

USAGE:
    rerobots search [FLAGS] [QUERY]

FLAGS:
    -h, --help                     Prints help information
    -V, --version                  Prints version information
        --include-user-provided
            include user_provided workspace deployments in search


ARGS:
    <QUERY>    

